### PR TITLE
Pre-RC updates

### PR DESCRIPTION
**Pull request by aburrell**
# Description

Addresses #1155 by comparing the package documents and tests to the ecosystem templates.  Also:
- Improves file handling in the package init,
- Fixes call to importlib's metadata,
- Updates the package citation, and
- Removes and old TODO statement

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# How Has This Been Tested?

Ran unit tests locally.  Also:

```
import pysat

pinst = pysat.Instrument('pysat', 'testing')
xinst = pysat.Instrument('pysat', 'ndtesting')
minst = pysat.Instrument('pysat', 'testmodel')
pinst.load(yr=2009, doy=1)
xinst.load(yr=2009, doy=1)
minst.load(yr=2009, doy=1)

# Printing 'longitude' for each instrument will show the range is 0-360. Change that
pysat.utils.coords.update_longitude(pinst, 'longitude')
pysat.utils.coords.update_longitude(xinst, 'longitude')
pysat.utils.coords.update_longitude(minst, 'longitude')

# Now -180-180
print(pinst['longitude'])
print(xinst['longitude'])
print(minst['longitude'])
```

**Test Configuration**:
* Operating system: OS X Big Sur
* Version number: Python 3.9
* Any details about your local setup that are relevant

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the release
checklist on the wiki: https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -60,7 +60,6 @@ test = [
   "flake8-docstrings",
   "hacking >= 1.0",
   "pysatSpaceWeather",
-  "pytest",
```
Comment (jklenzing) [inline pyproject.toml:63]: why is this removed?

Comment (aburrell) [reply pyproject.toml:63]: Because it is a part of the pysat requirements.  No need to request it twice.
